lib/pipefy_message/providers/aws_client/sns_broker.rb

```diff
@@ -27,18 +28,28 @@ def initialize(_opts = {})
         ##
         # Publishes a message with the given payload to the SNS topic
         # with topic_name.
-        def publish(payload, topic_name)
+        def publish(payload, topic_name, context = "NO_CONTEXT_PROVIDED")
           message = prepare_payload(payload)
           topic_arn = @topic_arn_prefix + (@is_staging ? "#{topic_name}-staging" : topic_name)
           topic = @sns.topic(topic_arn)
 
           logger.info(
             { topic_arn: topic_arn,
               payload: payload,
-              message_text: "Attempting to publish a json message to topic #{topic_arn}" }
+              message_text: "Attempting to publish a json message to topic #{topic_arn}}" }
           )
 
-          result = topic.publish({ message: message.to_json, message_structure: " json " })
+          result = topic.publish({ message: message.to_json, message_structure: " json ",
+                                   message_attributes: {
+                                     "correlationId" => {
+                                       data_type: "String",
+                                       string_value: SecureRandom.uuid.to_s
+                                     },
+                                     "context" => {
+                                       data_type: "String",
```

Comment:
I believe that `context` refers to `message_attributes`. Is it really a `String` or an `Object`?

Reply:
The intention of the `context` attribute is to allow the user to pass relevant information about the domain or flow that is using a given topic/queue, most of the time the topic/queue name doesn't express more than an action so I think that is a nice thing to have.

Also, is worth mentioning that it will be only used for tracing (logs) and visualization (dashboards) purposes.